ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,420 @@
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union
+
+import torch
+
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+
+def iou(y: torch.Tensor, y_pred: torch.Tensor, crowd: List) -> torch.Tensor:
+    m = len(y_pred)
+    n = len(y)
+
+    ious = torch.zeros(m, n)
+
+    for g in range(n):
+        Y = y[g].tolist()
```

Comment:
The naming of the `Y` variable does not respect the PEP8 convention. Please fix and check the file.